src/terminal/parser/InputStateMachineEngine.cpp

```diff
@@ -785,13 +733,9 @@ bool InputStateMachineEngine::_WriteMouseEvent(const size_t column, const size_t
 // - id - the identifier for the sequence we're operating on.
 // Return Value:
 // - the INPUT_RECORD compatible modifier state.
-DWORD InputStateMachineEngine::_GetCursorKeysModifierState(const gsl::span<const size_t> parameters, const VTID id) noexcept
+DWORD InputStateMachineEngine::_GetCursorKeysModifierState(const VTParameters parameters, const VTID id) noexcept
 {
-    DWORD modifiers = 0;
-    if (_IsModified(parameters.size()) && parameters.size() >= 2)
-    {
-        modifiers = _GetModifier(til::at(parameters, 1));
-    }
+    DWORD modifiers = _GetModifier(parameters.at(1));

```

Comment:
I'm not sure where on the barometer to place this, but it took me a little while to remember that _whether we produce a 0 or a 1_ is based on the receiving type. `_GetModifier` takes a size_t, so gets a 0. Discriminated types, as you called out, get a 1. If somebody were to change the type `_GetModifier` accepts, all this code would silently continue working (so long as the type it accepts meets `sizeof(T) == sizeof(size_t)`) but produce the wrong value.

Is that a concern? Or, am I overindexing on the concern?

Reply:
Okay, actually, I got the 0s and 1s backwards. I'm quite confused now -- does `_GetModifier` get a `0` here? Or it looks like it'll get a `1` because "omitted numeric parameters are 1 if 0 or omitted"? I'm confounded!

Reply:
Okay, I finally get it. No modifier is 1, presumably to support this exact behavior.

Reply:
Actually, I think you've caught a bug. This is one of the few cases where you've got a "numeric" parameter that shouldn't default to 1. If I'm reading the docs correctly, 0 is button 1 and 1 is button 2.

But this is why I have the `value_or` method, so we can override the default value when necessary. Ideally that'll be very rare, but it can be a problem with proprietary sequences that weren't developed by DEC. I still think it's worth have the default just work automatically 90% of the time.

Reply:
Looking into it a bit more... `_GetModifier` subtracts `1` as the first thing it does; I believe now that this _should_ default to 1. Otherwise modifier ends up being `-1` before we start pulling flags out of it.

Reply:
> If somebody were to change the type `_GetModifier` accepts, all this code would silently continue working

This is a valid concern, but I think in most cases this should be picked up by the unit tests. The input state machine doesn't have unit tests for stuff like this because it never supported default parameters before. I probably should add some.

Reply:
> Looking into it a bit more... `_GetModifier` subtracts `1` as the first thing it does

Yeah, sorry, I was looking at the wrong modifier. For some reason I thought we were talking about the mouse modifiers. But the mouse code already does the `value_or(0)` thing when processing the button parameter, so I think we're all good.